### PR TITLE
DAL, DI, clientside auth basics

### PR DESCRIPTION
**Pull request by AlexeyScherbina**

### CODE REVIEW

File: ScheduleTask.BLL/Services/TaskService.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+using ScheduleTask.DAL.Interfaces;
+using ScheduleTask.DAL;
+using ScheduleTask.DAL.Entities;
+
+
+namespace ScheduleTask.BLL.Services
+{
+    public class TaskService : ITaskService
+    {
+        IDataAccess Database { get; set; }
```
Comment (SKhalimonov) [inline ScheduleTask.BLL/Services/TaskService.cs]: Set as private

File: ScheduleTask.BLL/Services/UserService.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+using ScheduleTask.BLL.Interfaces;
+using ScheduleTask.DAL.Entities;
+using ScheduleTask.DAL.Interfaces;
+
+
+namespace ScheduleTask.BLL.Services
+{
+    public class UserService : IUserService
+    {
+        IDataAccess Database { get; set; }
```
Comment (SKhalimonov) [inline ScheduleTask.BLL/Services/UserService.cs]: Set as private

File: ScheduleTask.DAL/DataAccess.cs [hunk 3]
```diff
@@ -4,36 +4,36 @@
...
+
 
 namespace ScheduleTask.DAL
 {
     public class DataAccess : IDataAccess
     {
         private readonly ApplicationDbContext _db;
-        private TaskService _taskService;
-        private UserService _userService;
+        private TaskRepository _taskService;
```
Comment (SKhalimonov) [inline ScheduleTask.DAL/DataAccess.cs]: Rename to _taskRepository

File: ScheduleTask.DAL/DataAccess.cs [hunk 4]
```diff
@@ -4,36 +4,36 @@
...
         {
             _db = new ApplicationDbContext();
         }
-        public ITaskService Tasks
+        public ITaskRepository Tasks
         {
             get
             {
                 if (_taskService == null)
-                    _taskService = new TaskService(_db);
```
Comment (SKhalimonov) [inline ScheduleTask.DAL/DataAccess.cs]: Inject repository instead of creating

File: ScheduleTask.DAL/DataAccess.cs [hunk 5]
```diff
@@ -4,36 +4,36 @@
 using System.Text;
 using System.Threading.Tasks;
 using ScheduleTask.DAL.Interfaces;
-using Task1.Services;
+
 
 namespace ScheduleTask.DAL
 {
     public class DataAccess : IDataAccess
```
Comment (SKhalimonov) [inline ScheduleTask.DAL/DataAccess.cs]: Why did you create DataAccess? Im my opinion it's better to use repositories separately

File: ScheduleTask.DAL/Interfaces/TaskRepository.cs [hunk 6]
```diff
@@ -1,27 +1,30 @@
...
-namespace Task1.Services
+namespace ScheduleTask.DAL.Interfaces
 {
 
-    public interface ITaskService
+    public interface ITaskRepository
     {
         Tasks GetById(int id);
         List<Tasks> GetTasks();
         void AddTask(Tasks task);
```
Comment (IvanKrt) [inline ScheduleTask.DAL/Interfaces/TaskRepository.cs]: Why do wee need Word Task here? Why can`t we have base repository with all these actions like add/Get/getbyId/update/delete?

File: ScheduleTask.BLL/Services/UserService.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+        public UserService(IDataAccess db)
+        {
+            Database = db;
+        }
+
+        public void AddUser(UserDTO user)
+        {
+            User u = new User
```
Comment (IvanKrt) [inline ScheduleTask.BLL/Services/UserService.cs]: We should not use it this way. Saving data should be separately and using separate service

File: ScheduleTask.BLL/Services/TaskService.cs (hunk 1 above)
Comment (IvanKrt) [reply ScheduleTask.BLL/Services/TaskService.cs]: Was it resolved?

**Pull request merged**